File: erp-frontend/src/app/features/bitacora/bitacora.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { API_URL } from '../../core/api-config';
import { Page } from '../../core/models/page';
import {
  BitacoraAccesoResponse,
  BitacoraMovimientoResponse,
  FiltrosBitacoraAcceso,
  FiltrosBitacoraMovimiento,
} from './dto/bitacora';

const BASE_URL = `${API_URL}/bitacora`;

@Injectable({ providedIn: 'root' })
export class BitacoraService {
  constructor(private readonly http: HttpClient) {}

  listarMovimientos(filtros: FiltrosBitacoraMovimiento, page: number, size: number): Observable<Page<BitacoraMovimientoResponse>> {
    let params = new HttpParams().set('page', page).set('size', size);
    if (filtros.idUsuario) params = params.set('idUsuario', filtros.idUsuario);
    if (filtros.tabla) params = params.set('tabla', filtros.tabla);
    if (filtros.fechaDesde) params = params.set('fechaDesde', filtros.fechaDesde);
    if (filtros.fechaHasta) params = params.set('fechaHasta', filtros.fechaHasta);
    return this.http.get<Page<BitacoraMovimientoResponse>>(`${BASE_URL}/movimientos`, { params });
  }

  listarTablasConMovimientos(): Observable<string[]> {
    return this.http.get<string[]>(`${BASE_URL}/movimientos/tablas`);
  }

  listarAccesos(filtros: FiltrosBitacoraAcceso, page: number, size: number): Observable<Page<BitacoraAccesoResponse>> {
    let params = new HttpParams().set('page', page).set('size', size);
    if (filtros.idUsuario) params = params.set('idUsuario', filtros.idUsuario);
    if (filtros.resultado) params = params.set('resultado', filtros.resultado);
    if (filtros.fechaDesde) params = params.set('fechaDesde', filtros.fechaDesde);
    if (filtros.fechaHasta) params = params.set('fechaHasta', filtros.fechaHasta);
    return this.http.get<Page<BitacoraAccesoResponse>>(`${BASE_URL}/accesos`, { params });
  }
}

File: erp-backend/src/main/java/com/lacasadelchef/erp/administracion/auditoria/BitacoraService.java
package com.lacasadelchef.erp.administracion.auditoria;

import com.lacasadelchef.erp.administracion.auditoria.dto.BitacoraAccesoResponse;
import com.lacasadelchef.erp.administracion.auditoria.dto.BitacoraMovimientoResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.util.List;

public interface BitacoraService {

    Page<BitacoraMovimientoResponse> listarMovimientos(Integer idUsuario, String tabla,
                                                         LocalDate fechaDesde, LocalDate fechaHasta, Pageable pageable);

    Page<BitacoraAccesoResponse> listarAccesos(Integer idUsuario, String resultado,
                                                LocalDate fechaDesde, LocalDate fechaHasta, Pageable pageable);

    /** Tablas que aparecen en bitacora_movimiento, para poblar el filtro de la pantalla. */
    List<String> listarTablasConMovimientos();
}
